expirationd.lua

```diff
@@ -359,21 +384,131 @@ local function expirationd_run_task(name, space_id, is_tuple_expired, options)
+    --            format[ part.fieldno ].type = part.type
+    --        end
+    --    end
+    --
+    --    local expire_field_no
+    --    if have_format and type(options.field) == "string" then
+    --        expire_field_no = format[ options.field ].no
+    --    else
+    --        expire_field_no = options.field
+    --    end
+    --    if type(expire_field_no) ~= 'number' then
+    --        error("Need correct field option")
+    --    end
+    --
+    --    for _, index in pairs(box.space[space_id].index) do
+    --        -- we use only first part of index,
+    --        -- because we will have problems with the starting element
+    --        -- perhaps we should first of all take an index consisting only of our field
+    --        if index.parts[1].fieldno == expire_field_no then
+    --            expire_index = index
+    --        end
+    --    end
+    --    expire_index = expire_index
+    --else
+    --    -- take default primary index
+    --    expire_index = box.space[space_id].index[0]
+    --end
+    --task.expire_index = expire_index
+
+    -- validate index
+    local expire_index = box.space[space_id].index[0]
+    if options.index then
+        local index_exists = false
+        for _, index in pairs(box.space[space_id].index) do
+            if index.name == options.index then
+                expire_index = index
+                index_exists = true
+            end
+        end
+        if not index_exists then
+            error("Index with name " .. options.index .. " not exists")
+        end
+    end
+    task.expire_index = expire_index
+
+    -- check ascending
+    if options.ascending ~= nil then
+        if type(options.ascending) ~= 'boolean' then
+            error("Invalid type of ascending, expected boolean")
+        end
+        if not options.ascending then
+            task.iterator = "LE"
+        end
+    end
+
+    if options.start_element then
+        if type(options.start_element) == "table" then
+            if #expire_index.parts ~= #options.start_element then
+                error("Wrong number of elements in start_element table, expected: " .. #expire_index.parts)
+            end
+            for i = 1, #expire_index.parts do
+                log.info(expire_index.parts[i].type)
+                if expire_index.parts[i].type ~= type(options.start_element[i]) then
+                    error("Wrong type in " .. i ..
+                             " element of start_element table, expected: " .. expire_index.parts[i].type ..
+                             ", got: " .. type(options.start_element[i]))
+                end
+            end
+        else
+            if #expire_index.parts ~= 1 then
+                error("Wrong number of elements in start_element table, expected: " .. #expire_index.parts)
+            end
+            if expire_index.parts[1].type ~= type(options.start_element) then
```

Comment:
There will be an error if the index type is unsigned, and in options we accept number e.g. 123

Reply:
As well for 'scalar' index part type, 'decimal'.

Reply:
Made the check by calling `index:pairs (key, iterator)`, as suggested by @olegrok